src/EditorFeatures/Core/InlineHints/InlineHintsDataTaggerProvider.cs

```diff
@@ -111,7 +95,7 @@ protected override async Task ProduceTagsAsync(
 
         var options = GlobalOptions.GetInlineHintsOptions(document.Project.Language);
 
-        var snapshotSpan = documentSnapshotSpan.SnapshotSpan;
+        var snapshotSpan = spanToTag.SnapshotSpan;
```

Comment:
Couple nits not related to this PR, I'm not sure how often this code is invoked:

1) Should the code in the loop be using Text instead of ToString()? I know the impl delegates ToString() to Text, but it seems weird to rely on that.

2) The Sum could instead by an All with length 0. No need to walk all the display parts.